Cython/Utility/arrayarray.h

```diff
@@ -105,9 +106,20 @@ PyObject* newarrayobject(PyTypeObject *type, Py_ssize_t size,
     struct arraydescr *descr);
 #endif /* ifndef NO_NEWARRAY_INLINE */
 
+static CYTHON_INLINE __data_union __Pyx_PyArray_Data(arrayobject *self) {
+#if !CYTHON_COMPILING_IN_GRAAL
+    return self->data;
+#else
+    __data_union data;
+    data.ob_item = GraalPyArray_Data((PyObject*)self);
+    return data;
+#endif
+}
+
 // fast resize (reallocation to the point)
 // not designed for filing small increments (but for fast opaque array apps)
 static CYTHON_INLINE int resize(arrayobject *self, Py_ssize_t n) {
+#if !CYTHON_COMPILING_IN_GRAAL
```

Comment:
```suggestion
#if CYTHON_COMPILING_IN_GRAAL
    return GraalPyArray_Resize((PyObject*)self, n);
#else
```

Similarly here moving the general case to the else